airflow/gcp/hooks/bigquery.py

```diff
@@ -1109,6 +1110,16 @@ def run_table_upsert(self, dataset_id: str, table_resource: Dict,
         """
         service = self.get_service()
         # check to see if the table exists
+        if 'tableReference' not in table_resource:
+            raise AirflowBadRequest(
+                '"tableReference" is required within table_resource parameter. '
+                'See https://cloud.google.com/bigquery/docs/reference/v2/tables#resource'
+            )
+        if 'tableId' not in table_resource["tableReference"]:
+            raise AirflowBadRequest(
+                '"tableId" is required within table_resource["tableReference"]. '
+                'See https://cloud.google.com/bigquery/docs/reference/v2/tables#resource'
+            )
```

Comment:
I just wonder, does it break backward compatibility?

Reply:
@nuclearpinguin If your concern is about the operators then I think we are safe because currently none of the operators are using this method. The only calls are from test cases.

Memo:
```
:~/Desktop/search_work/airflow|master 
⇒  git grep run_table_upsert

CHANGELOG.txt:- [AIRFLOW-314] Fix BigQuery cursor run_table_upsert method
CHANGELOG.txt:- Add two methods to bigquery hook's base cursor: run_table_upsert, which adds a table or updates an existing table; and run_grant_dataset_view_access, which grants view access to a given dataset for a given table.
UPDATING.md:| airflow.gcp.hooks.bigquery.BigQueryBaseCursor.run_table_upsert              | airflow.gcp.hooks.bigquery.BigQueryHook.run_table_upsert              |
airflow/gcp/hooks/bigquery.py:    def run_table_upsert(self, dataset_id: str, table_resource: Dict,
airflow/gcp/hooks/bigquery.py:    def run_table_upsert(self, *args, **kwargs) -> Dict:
airflow/gcp/hooks/bigquery.py:        Please use `airflow.gcp.hooks.bigquery.BigQueryHook.run_table_upsert`
airflow/gcp/hooks/bigquery.py:            "Please use `airflow.gcp.hooks.bigquery.BigQueryHook.run_table_upsert`",
airflow/gcp/hooks/bigquery.py:        return self.hook.run_table_upsert(*args, **kwargs)
tests/gcp/hooks/test_bigquery.py:        bq_hook.run_table_upsert(dataset_id=DATASET_ID, table_resource=table_resource)
tests/gcp/hooks/test_bigquery.py:        bq_hook.run_table_upsert(dataset_id=DATASET_ID, table_resource=table_resource)
tests/gcp/hooks/test_bigquery.py:        ("run_table_upsert",),
```

Reply:
Users can use this method in their own custom operators that are not a part of Airflow. In this case, we have to be sure that backward compatibility is preserved... especially because it's a BQ which is quite popular in Airflow.

Reply:
I see... Well even though these are quite trivial tests and google API would any fail for such requests.
But still, let me remove them for now.

Reply:
@nuclearpinguin removed https://github.com/apache/airflow/pull/7126/commits/e735ac1438970f333c8c3c681e9f93d5543021f9 and rebased 🙇